Support Python 3: initial ops

adam-thomas opened the pull request:
Expanding the test coverage (see https://github.com/incuna/django-admin-sso/issues/4) and adding Python 3 support to the untested stretches of the code go hand in hand, so I can't really add Python 3 support all in one go (unless it miraculously works) with one issue.  This issue is about P3 support before I expand tests - i.e. when the initial six tests in test_auth all pass on Python 2 and 3, this can be reviewed and merged.

adam-thomas committed:
Update .travis.yml to use Python 3.

* Python 3 and Django < 1.5 are incompatible, so we'll
  need to exclude those combinations.

adam-thomas commented:
Don't merge yet - I'm using the issue so I can get Travis to run tests for me :)

meshy commented on .travis.yml:
```diff
@@ -2,10 +2,19 @@ language: python
+  exclude:
+     - env: DJANGO_VERSION<1.5
+       python: 3.3
+     - env: DJANGO_VERSION<1.5
```
I think this will need to be `DJANGO_VERSION="django>=1.4,<1.5"`

meshy replied:
Also, I think your multi-line yaml syntax may be wrong

meshy replied:
(maybe)

adam-thomas replied:
I think the syntax is ok (I copied it out of another file), but needing it to be `DJANGO_VERSION="django>=1.4,<1.5"` appears to be the case.

meshy replied:
Cool :+1:

adam-thomas commented:
![screenshot from 2014-05-19 10 59 45](https://cloud.githubusercontent.com/assets/6973481/3012661/5acc87fa-df3c-11e3-93a6-53a7eb90a49b.png)

Excellent.  Let's get started.

meshy commented:
![https://i.chzbgr.com/maxW500/4783401216/hED3700B5/](https://i.chzbgr.com/maxW500/4783401216/hED3700B5/)

adam-thomas committed:
Use python3-openid and fix deprecations.

* openid isn't compatible with P3, but some merciful soul
  has made a version of it that works. Change setup.py
  to install the correct version (and only the correct
  version).
* test_auth uses assertEquals a bunch of times; change
  that to the non-deprecated assertEqual.

adam-thomas committed:
Base64 and openid compatibility tweaks.

* encodestring and decodestring are outdated, so swap them for
  b64encode and b64decode.
* python-openid and python3-openid have a different name for
  [get]ExpiresIn, so we need to do some hacking to ensure
  compatibility.

meshy commented on admin_sso/store.py:
```diff
@@ -71,11 +71,21 @@ def getAssociation(self, server_url, handle=None):
         expired = []
         for assoc in assocs:
+            print(assoc.secret)
+            print(type(assoc.secret))
```
Couple of stray `print`s here `;)`

adam-thomas replied:
...oops :P

adam-thomas commented:
Made a pull request to the real deal repository (after a rebase to tidy up the history) at https://github.com/frog32/django-admin-sso/pull/12.

adam-thomas committed:
Allow a little flexibility in OpenID distributions.

adam-thomas committed:
Fix typo.

adam-thomas commented:
Closing this since the replacement fork at https://github.com/matthiask/django-admin-sso supports Python 3.
